enclave/src/test/cert_tests.rs

```diff
@@ -0,0 +1,75 @@
+
+pub fn test_verify_mra_cert_should_work() {
+	let mr_enclave = get_mr_enclave_from_hex_string(TEST4_MRENCLAVE).unwrap();
+	let attestation_ocall =
+		AttestationOCallMock::create_with_mr_enclave(sgx_measurement_t { m: mr_enclave });
+	let result = verify_mra_cert(TEST4_CERT, &attestation_ocall);
+
+	assert!(result.is_ok());
+}
+
+pub fn test_verify_wrong_cert_is_err() {
+	let mr_enclave = get_mr_enclave_from_hex_string(TEST4_MRENCLAVE).unwrap();
+	let attestation_ocall =
+		AttestationOCallMock::create_with_mr_enclave(sgx_measurement_t { m: mr_enclave });
+	let result = verify_mra_cert(CERT_WRONG_PLATFORM_BLOB, &attestation_ocall);
+
+	assert!(result.is_err());
+	assert_eq!(result.unwrap_err(), sgx_status_t::SGX_ERROR_UNEXPECTED);
+}
+
+pub fn test_given_wrong_platform_info_when_verifying_attestation_report_then_return_error() {
+	let attestation_ocall = AttestationOCallMock::new();
+	let result = verify_attn_report(CERT_WRONG_PLATFORM_BLOB, Vec::new(), &attestation_ocall);
+
+	assert!(result.is_err());
+	assert_eq!(result.unwrap_err(), sgx_status_t::SGX_ERROR_UNEXPECTED);
+}
```

Comment:
These tests and their corresponding fixtures have been more or less copied from https://github.com/integritee-network/pallet-teerex/blob/master/ias-verify/src/lib.rs